### PR TITLE
Implement max comments feature. Add delete all comments button.

### PR DESCRIPTION
**Pull request by egrimshaw**
I adapted my JS functions to include one that will set the max number of comments to be displayed when the submit limit button is clicked, and another that will load the amount of comments from all comments inputted (sorted by most recent first) based on that amount set before hand. I did this through session storage. I also added a button that will remove all comments from the datastore and delete them all from the page. 

I apologize if there are any print statement or commented out code remaining. I cleaned it up, but I may have missed something. 

Next, I am going to tackle allowing the user to delete a specific comment rather than all of them. 

Thank you!

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  private final long id;
+  private final String comment;
+  private final long time;
+
+  public Comment(long id, String commentAdd, long timestamp) {
+    this.id = id;
+    this.comment = commentAdd;
+    this.time = timestamp;
+  }
+}
```
Comment (arjun-garg-google) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:15]: nit: add a newline at the end of the file

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -55,19 +55,42 @@ function createList(text){
...
         commentElement.appendChild(olTag);
+       
     });
 
 }
 
+//This function deletes all comments.
+async function deleteComments(){
+  await fetch('/delete-data', {method: 'POST'});
+  fetch('/load').then(response => response.json()).then((arrayString) =>
```
Comment (arjun-garg-google) [inline portfolio/src/main/webapp/script.js:89]: best to be consistent, if you're going to use async await, use it for all the promises in the function

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
+
+/** Servlet that loads some comments content. */
+@WebServlet("/delete-data")
+public class DeleteData extends HttpServlet {
+
+  private List<String> commentArray = new ArrayList<String>();
```
Comment (arjun-garg-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:41]: doesn't look like this is used anywhere

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comments").addSort("time", SortDirection.DESCENDING); // most recent comments first
```
Comment (arjun-garg-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:45]: do you need to sort by time here?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -47,14 +47,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("viewer-comment");
     String commentName = request.getParameter("viewer-comment-name");
-    String commentCombined = "\"" + comment + "\"" + " -" + commentName;
+    String commentCombined = null;
+    if (comment!=null){ //don't create a comment unless there is text in the comment box
+        if (commentName==null){
+            commentName = "Annoymous";
+        }
+     commentCombined = "\"" + comment + "\"" + " -" + commentName;
```
Comment (arjun-garg-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: Is there a reason to do this vs just having another property on the comment that is the comment name?

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java [hunk 6]
```diff
@@ -38,19 +40,32 @@
...
+      Comment commentToAdd = new Comment(id, title, timestamp);
+      commentsArray.add(commentToAdd);
+      
     }
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(commentsArray));
 
+
```
Comment (arjun-garg-google) [inline portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:69]: nit: extra newline?

File: portfolio/src/main/webapp/commentform.html [hunk 7]
```diff
@@ -9,18 +9,30 @@
...
-  <body class = "main-body" onload = "loadComments()">
+  <body class = "main-body" onload = "loadAllComments()">
         <div id = "nav"> </div>
         <script> $(document).ready(function(){$("#nav").load("navbar.html")}); </script>
         <form action = "/data" method = "POST">
-            <p id = "commentText"> Comment: </p>
+            <p class = "commentText"> Enter a number between 0 and 20 for the number of comments 
+                you would like to be displayed: </p>
+            <input type = "number" name = "numComments" id = "max" min = "0" max = "20" value = "1" >
+            <br>
```
Comment (arjun-garg-google) [inline portfolio/src/main/webapp/commentform.html:19]: nit: looks like there is mixed usage of <br> and <br/> here. Try to use <br> everywhere

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java (hunk 4 above)
Comment (egrimshaw) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:45]: Hi! I don't need to, but I decided to sort it like that as most of the comment pages I've seen on websites show most recent postings first. I can swap this around or delete any sorting though.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -47,14 +47,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("viewer-comment");
     String commentName = request.getParameter("viewer-comment-name");
-    String commentCombined = "\"" + comment + "\"" + " -" + commentName;
+    String commentCombined = null;
```
Comment (arjun-garg-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: I think you can get rid of this variable

**Pull request merged**